main.js

```diff
@@ -0,0 +1,250 @@
+}
+
+console.log(getVoteCount({ upvotes: 10, downvotes: 5 }));
+console.log(getVoteCount({ upvotes: 75, downvotes: 90 }));
+console.log(getVoteCount({ upvotes: 50, downvotes: 50 }));
+
+// 7
+
+function getCubeVolume(cube) {
+    return cube.height * cube.length * cube.width;
+}
+
+console.log(getCubeVolume({ width: 10, length: 5, height: 2 }));
+console.log(getCubeVolume({ width: 100, length: 500, height: 0 }));
+console.log(getCubeVolume({ width: 15, length: 2, height: 5 }));
+
+// 8
+
+function getCityInformation(city) {
+    return `${city.name} is in ${city.country} and has an area of ${city.areaInKilometers} square kilometers`;
+}
+
+console.log(getCityInformation({
+    name: 'Warsaw',
+    country: 'Poland',
+    areaInKilometers: 517
+}));
+
+// 9
+
+const john = {
+    firstName: 'John',
+    lastName: 'Smith'
+}
+
+function getUserCopy(user) {
+    return {
+        firstName: user.firstName,
+        lastName: user.lastName
+    }
+}
+
+const newUser = getUserCopy(john);
+console.log(newUser.firstName);
+console.log(newUser.lastName);
+console.log(newUser === john);
+
+// 10
+
+function getAnimalDescription(animal){
+    return `This ${animal.color} ${animal.name} has ${animal.legs} legs.`
+}
+
+// 11
+
+const rooms = {
+    first: {
+        description: 'This is the first room',
+        items: {
+            chair: 'The old chair looks comfortable',
+            lamp: 'This lamp looks ancient'
+        }
+    },
+    second: {
+        description: 'This is the second room',
+        items: {
+            couch: 'This couch looks like it would hurt your back',
+            table: 'On the table there is an unopened bottle of water'
+        }
+    }
+}
+
+// 12
+
+function addContact(contacts, name, phoneNumber) {
+    const newContacts = {...contacts};
+
+    newContacts[name] = {
+        name: name,
+        phoneNumber: phoneNumber
+    }
+
+    return newContacts;
+}
+
+const contacts = {};
+const contactsWithJohn = addContact(contacts, 'John', '123 456 789');
+
+console.log(contactsWithJohn.John.name);
+console.log(contactsWithJohn.John.phoneNumber);
+console.log(contacts === contactsWithJohn);
+console.log(contacts.John);
+
+// 13
+
+function findContact(object, property) {
+    if (object.hasOwnProperty(property)) {
+        return `Name: ${property}, Phone: ${object[property].phoneNumber}`;
+    }
+    return;
+}
+
+const johnContactInfo = findContact(contactsWithJohn, 'John');
+console.log(johnContactInfo);
+
+// 14
+
+const userThree = {
+    firstName: 'Bob',
+    lastName: 'Ross',
+    age: 40,
+    greet() {
+        return "Hi, I'm Bob Ross!";
+    }
+}
+const objectExample = {
+    firstName: 'Kate',
+    lastName: 'Williams',
+    age: 40,
+    greet() {
+        return "Hi, I'm Kate!";
+    }
+}
+const car = {
+    make: 'Ferrari',
+    model: 'F40',
+    greet() {
+        return "<engine noise>";
+    }
+}
+
+function validateIfFirstNameisString(object) {
+    return object.hasOwnProperty('firstName')
```

Comment:
You can simplify this a lot with just `typeof object.firstName === 'string'`. If the object does not have this property, its type will be undefined.